### PR TITLE
Adding run functionality

### PR DESCRIPTION
**Pull request by joelridden**
Allowing addition for adding new runs to the database.
Changes include allowing for Historical run types that can be stored elsewhere to the Cybershake Dropbox directory.
While adding a run you get notified of errors such as an incorrect secret key so only the software team can add runs to the DB. (Will generate a secret key that will actually be used later once its deployed)
Also informs if there was an error that occurred during the adding of the run and informs of a success too.
image
image
image

### CODE REVIEW

File: cs_api/cs_api/api/runs.py [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 import flask
+import pandas as pd
 from flask_cors import cross_origin
 
 from cs_api import server, utils
 from cs_api import constants as const
-from cs_api.db import db
+from cs_api.db import db as cs_db
+from cs_api.server import db
+from cs_api.db.models import *
```
Comment (claudio525) [inline cs_api/cs_api/api/runs.py]: Not sure about `*` import

File: cs_api/cs_api/api/runs.py [hunk 2]
```diff
@@ -88,3 +96,90 @@ def get_runs_from_interests():
...
+        site_df[run_name] = False
+
+        # Create a temporary directory to get the site information for the run
+        temp_dir = TemporaryDirectory()
+        temp_path = Path(temp_dir.name).resolve()
+
+        cs_dropbox_download_py_ffp = os.environ["CS_DROPBOX_DOWNLOAD_PY_FFP"]
+
+        # Download the IM data
+        p = subprocess.Popen(
```
Comment (claudio525) [inline cs_api/cs_api/api/runs.py]: Any particular reason for calling python in a subprocess and not just calling it directly?

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py [hunk 3]
```diff
@@ -73,6 +80,7 @@ def download(
     dropbox_cs_ver: str,
     data_types: list,
     download_root: Path,
+    dropbox_directory: str = None,
```
Comment (sungeunbae) [inline dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py:83]: Just a thought : can dropbox_directory be a Path?  Does it make things more tedious to convert back to str()?

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py [hunk 4]
```diff
@@ -197,6 +210,7 @@ def download(
             continue
 
         print(f"### Extracting {t}")
+        tar = tarfile.open(t, "r")
```
Comment (sungeunbae) [inline dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py]: Do you use this?

**Review by sungeunbae — COMMENTED** [review]
I have a few questions : not critical

File: cs_api/cs_api/api/runs.py (hunk 2 above)
Comment (joelridden) [reply cs_api/cs_api/api/runs.py]: Very fair point, forgot I made a function to do this easier

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py (hunk 3 above)
Comment (joelridden) [reply dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py:83]: Well the dropbox directory variable contains the "dropbox:" part so it's not really a Path. Don't see any reason to convert a string from the API request to a Path to then a string again, simpler to just keep it a string in my opinion.

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py (hunk 4 above)
Comment (joelridden) [reply dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py]: Doesn't look like it, seems to be something I missed when having to do a merge conflict.

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py (hunk 3 above)
Comment (sungeunbae) [reply dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_download.py:83]: OK, sounds like you thought about it while you were implementing it. Cool.

**Review by sungeunbae — APPROVED** [review]

**Comment by sungeunbae** [comment]
@joelridden Oh, what should we do with this PR?

**Comment by joelridden** [comment]
> @joelridden Oh, what should we do with this PR?

I think I actually have this code in the PR for the new repo, seems to be tidied to a good state so thats good, just missed that this existed.

**Pull request closed without merging**